cetmix_tower_server/wizards/cx_tower_server_template_create_wizard.py

```diff
@@ -144,7 +156,48 @@ class CxTowerServerTemplateCreateWizardVariableLine(models.TransientModel):
     wizard_id = fields.Many2one("cx.tower.server.template.create.wizard")
     variable_id = fields.Many2one(comodel_name="cx.tower.variable", required=True)
     variable_reference = fields.Char(related="variable_id.reference", readonly=True)
-    value_char = fields.Char(string="Value")
+    value_char = fields.Char(string="Value", readonly=False)
     required = fields.Boolean(
         help="Indicates if this variable is mandatory for server creation",
     )
+    variable_type = fields.Selection(
+        related="variable_id.variable_type",
+        readonly=True,
+    )
+    option_id = fields.Many2one(
+        comodel_name="cx.tower.variable.option", ondelete="restrict"
+    )
+    option_ids_domain = fields.Binary(compute="_compute_option_ids_domain")
+
+    @api.depends("option_id", "variable_id.option_ids")
+    def _compute_option_ids_domain(self):
+        """
+        Compute the domain for the `option_ids_domain` field based on the related
+        `option_id` and the `option_ids` of the associated `variable_id`.
+        """
+        for rec in self:
+            allowed_option_ids = rec.variable_id.option_ids.ids
+            rec.option_ids_domain = [("id", "in", allowed_option_ids)]
+
+    @api.onchange("option_id", "variable_id.option_ids", "value_char")
+    def _onchange_value_char(self):
+        for rec in self:
+            if rec.variable_id.option_ids and rec.option_id:
+                rec.value_char = rec.option_id.value_char
+            elif not rec.variable_id.option_ids:
+                rec.value_char = rec.value_char or ""
+                rec.option_id = None
+                rec.variable_type = "s"
+
+    @api.onchange("variable_id")
+    def _onchange_variable_id(self):
+        """
+        Reset option_id when variable changes or
+        doesn't have options
+        """
+        for rec in self:
+            if rec.variable_id.option_ids:
+                allowed_option_ids = rec.variable_id.option_ids.ids
+                rec.option_ids_domain = [("id", "in", allowed_option_ids)]
```

Comment:
You already have a function that computes the domain. Besides domain fields is readonly so you cannot update it manually.